### PR TITLE
feat: enable preview of Fedora 40 kmod builds

### PR DESCRIPTION
**Pull request by bsherman**
This enables Fedora 40 builds, but filters out known problems.

Specifically, when the build version is Fedora 40, this disables several kmods which are either not yet available for Fedora 40 in upstream repos or do not currently compile against Fedora 40.

The disabled kmod list:
- evdi: negativo17 does not yet provide F40
- gasket: compile failure
- nvidia-470xx: compile failure
- rtl18814au: compile failure
- rtl188xxau: compile failure
- v4l2loopback: compile failure
- xpadneo: negativo17 does not yet provide F40

I also standardized a small change to the build-kmod scripts. We now copy/activate extra yum repo files only after getting the Fedora release variable which make things simpler when needing to bail out early due to known issues with F40, etc.

### CODE REVIEW

**Review by p5 — APPROVED** [review]
LGTM once the builds are green.

Created a tracking issue so we can monitor the rollout to the remaining akmods
https://github.com/ublue-os/akmods/issues

**Comment by KyleGospo** [comment]
We can likely drop xpadneo entirely. Xone and xpad should cover 99% of the use case, Bazzite for instance doesn't ship xpadneo.

I will look into Gasket since that's my pet

**Comment by bsherman** [comment]
> We can likely drop xpadneo entirely. Xone and xpad should cover 99% of the use case, Bazzite for instance doesn't ship xpadneo.
> 
> I will look into Gasket since that's my pet

xpadneo is the bluetooth support for xone ... what do you use instead?

**Comment by bsherman** [comment]
FYI all approvers, I'm fixing more workflow stuff as the PR checks build has shown some gaps in my local testing.

**Comment by KyleGospo** [comment]
> > We can likely drop xpadneo entirely. Xone and xpad should cover 99% of the use case, Bazzite for instance doesn't ship xpadneo.
> > 
> > I will look into Gasket since that's my pet
> 
> xpadneo is the bluetooth support for xone ... what do you use instead? 
> 
> edit: read-between the lines :-) I use xpadneo a lot

We only ship xpad and xone and haven't had complaints. I think Steam does something in user space.

**Comment by bsherman** [comment]
> > xpadneo is the bluetooth support for xone ... what do you use instead?
> 
> We only ship xpad and xone and haven't had complaints. I think Steam does something in user space.

Fair enough. I'm open to considering removal as part of another discussion where we test it out, but I'd rather not make the decision as part of this PR.

**Comment by KyleGospo** [comment]
We only ship xpad and xone and haven't had complaints. I think Steam does something in user space.

**Comment by bsherman** [comment]
Apparently I broke something in the F38/F39 common akmod builds. :fearful: 

Trying to track that down and fix it, then this should be good to go with the caveats listed above  and  in the related #144

**Comment by bsherman** [comment]
Ok! This is looking good now!

**Review by castrojo — APPROVED** [review]

**Pull request merged**
